tests/jms-tests/src/test/java/org/apache/activemq/artemis/jms/tests/tools/container/InVMInitialContextFactory.java

```diff
@@ -104,10 +100,6 @@ public Context getInitialContext(final Hashtable<?, ?> environment) throws Namin
       }
    }
 
-   public static void reset() {
-      InVMInitialContextFactory.initialContexts = new HashMap<>();
```

Comment:
This is a public method we shouldnt be removing.

Reply:
Nobody seems to be calling that, and it is in the tests part of the codebase. If we want to keep it, then maybe calling `initialContexts.clear()` (under whatever synchronization that is possibly required) would be sufficient.

If it needs further thought, I can drop it from the PR.

Reply:
As long as no ones using this in extended tests then fair enough. Concern is just where people may use and extend it in their own projects. If maybe cross check with wildfly project (known big user) to make sure. Then im happy